src/controller/CHIPDeviceController.cpp

```diff
@@ -489,6 +489,17 @@ void DeviceCommissioner::Shutdown()
     }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
 
+    // If we have a commissionee device for the device being commissioned,
+    // release it now, before we release our whole commissionee pool.
+    if (mDeviceBeingCommissioned != nullptr)
+    {
+        auto * commissionee = FindCommissioneeDevice(mDeviceBeingCommissioned->GetDeviceId());
```

Comment:
Is this step needed because the commissionee device proxy may not be allocated from the common pool? If that's possible, shouldn't the code for releasing `mDeviceInPASEEstablishment` also check that? And if that's not possible, wouldn't `ReleaseCommissioneeDevice(mDeviceBeingCommissioned)` be enough? Or even `mDeviceBeingCommissioned = nullptr` after `mCommissioneeDevicePool.ReleaseAll()`?

Reply:
> Is this step needed because the commissionee device proxy may not be allocated from the common pool?

`mDeviceBeingCommissioned` might be an operational device, not a commissionee device, if we got to the "send CommissioningComplete" part of commissioning.  That's why it's a `DeviceProxy *`, not `CommmissioneeDeviceProxy *`.

> wouldn't ReleaseCommissioneeDevice(mDeviceBeingCommissioned)

Wouldn't compile; wrong type.

> Or even mDeviceBeingCommissioned = nullptr after mCommissioneeDevicePool.ReleaseAll()

Again, `mDeviceBeingCommissioned` might not be a `CommmissioneeDeviceProxy`.

Reply:
Oh, ok. Wasn't aware it can be operational device, too. Thanks!